server/src/main/java/io/crate/expression/operator/any/AnyLikeOperator.java

```diff
@@ -56,7 +57,10 @@ public AnyLikeOperator(Signature signature,
 
     @Override
     boolean matches(Object probe, Object candidate) {
-        return LikeOperators.matches((String) candidate, (String) probe, caseSensitivity);
+        // Accept both sides of arguments to be patterns
+        // since CrateDB originally supported `'%bar' like any (['foobar', 'bar'])`
```

Comment:
minor, but I think you can remove this 2nd line, I wouldn't mention the "history" of functionality with a comment on the code.